src/commands/CmdCalendar.cpp

```diff
@@ -606,6 +610,25 @@ std::string CmdCalendar::renderMonths (
         // colorize due tasks
         if (config.get ("calendar.details") != "none")
         {
+          config.set ("schdeuled", 0);
+          for (auto& task : all)
+          {
+            auto status = task.getStatus ();
+            if ((status == Task::pending ||
+                 status == Task::waiting ) &&
+                !task.hasTag ("nocal")     &&
+                task.has ("scheduled"))
+            {
+              std::string due = task.get ("scheduled");
+              Datetime duedmy (strtol (due.c_str(), nullptr, 10));
+
+              if (duedmy.sameDay (date))
+              {
+                cellColor.blend(color_scheduled);
+              }
+            }
+          }
+
```

Comment:
I think it would be a bit less repetitive here if we could roll the check for scheduled attribute into the code below. In particular:

```c++
            auto status = task.getStatus ();
            if ((status == Task::pending ||
                 status == Task::waiting ) &&
                !task.hasTag ("nocal")     &&
                task.has ("due"))
```

could be altered to omit the last condition and then we could split the logic within one cycle depending on whether "due" is defined or "scheduled" is defined (or both).

Reply:
Hi,
this is obviously correct - I changed the code as youu proposed. I switched the order of the due and scheduled section, since I think if a task is due on that day it is more important then when one is scheduled. Also, I added a flag that once a due task has been found on that day, no coloring of scheduled tasks will happen.